private/bufpkg/bufplugin/bufpluginconfig/config.go

```diff
@@ -100,92 +105,73 @@ func newRuntimeConfig(externalRuntimeConfig ExternalRuntimeConfig) (*RuntimeConf
+	// that this will change if/when there are more runtime languages supported.
+	goRuntimeConfig, err := newGoRuntimeConfig(externalRuntimeConfig.Go)
+	if err != nil {
+		return nil, err
 	}
-	// If we made it this far, that means the config specifies multiple
-	// runtime languages.
-	//
-	// We might eventually want to support multiple runtime configuration
-	// (e.g. 'go' and 'archive'), but it's safe to start with an error for
-	// now.
-	return nil, fmt.Errorf("%s configuration contains multiple runtime languages", ExternalConfigFilePath)
+	return &RuntimeConfig{
+		Go: goRuntimeConfig,
+	}, nil
 }
 
 func newNPMRuntimeConfig(externalNPMRuntimeConfig ExternalNPMRuntimeConfig) (*NPMRuntimeConfig, error) {
-	if err := validateRuntimeDeps(externalNPMRuntimeConfig.Deps); err != nil {
-		return nil, err
+	var dependencies []*NPMRuntimeDependencyConfig
+	for _, dep := range externalNPMRuntimeConfig.Deps {
+		if dep.Package == "" {
+			return nil, errors.New("npm runtime dependency requires a non-empty package name")
+		}
+		if dep.Version == "" {
+			return nil, errors.New("npm runtime dependency requires a non-empty version name")
+		}
+		// TODO: Note that we don't have NPM-specific validation yet - any
+		// non-empty string will work for the package and version.
+		//
+		// For a complete set of the version syntax we need to support, see
+		// https://docs.npmjs.com/cli/v6/using-npm/semver
+		//
+		// https://github.com/Masterminds/semver might be a good candidate for
+		// this, but it might not support all of the constraints supported
+		// by NPM.
+		dependencies = append(
+			dependencies,
+			&NPMRuntimeDependencyConfig{
+				Package: dep.Package,
+				Version: dep.Version,
+			},
+		)
 	}
 	return &NPMRuntimeConfig{
-		Deps: externalNPMRuntimeConfig.Deps,
+		Deps: dependencies,
 	}, nil
 }
 
 func newGoRuntimeConfig(externalGoRuntimeConfig ExternalGoRuntimeConfig) (*GoRuntimeConfig, error) {
-	if err := validateRuntimeDeps(externalGoRuntimeConfig.Deps); err != nil {
-		return nil, err
-	}
-	// The best we can do is verify that the minimum version
-	// is a valid semantic version, just like we do for the
-	// runtime dependencies.
-	//
-	// This will not actually verify that the go version is
-	// in the valid set. It's impossible to capture the
-	// real set of valid identifiers at any given time (for
-	// an old version of the buf CLI) without reaching out to
-	// some external source at runtime.
-	//
-	// Note that this ensures the user's configuration specifies
-	// a 'v' prefix in the version (e.g. v1.18) even though the
-	// minimum version is rendered without it in the go.mod.
-	if externalGoRuntimeConfig.MinVersion != "" && !semver.IsValid(externalGoRuntimeConfig.MinVersion) {
+	if externalGoRuntimeConfig.MinVersion != "" && !modfile.GoVersionRE.MatchString(externalGoRuntimeConfig.MinVersion) {
 		return nil, fmt.Errorf("the go minimum version %q must be a valid semantic version", externalGoRuntimeConfig.MinVersion)
```

Comment:
Do we want to be more specific here - this will allow only `<major>.<minor>` syntax: https://pkg.go.dev/golang.org/x/mod/modfile#pkg-variables